### PR TITLE
Fix backend error

### PR DESCRIPTION
**Pull request by snorrees**
## Reviewer questions

I just commited this as patch-changes, but since the types will now "work" this _could_ break existing integrations.
Should this be a breaking change?

## Reported error

Hierarchical input: "Cannot have two HTML 5 backends"
Knut reported errors with the plugin.

"Upgraded to 1.0 and got an error when I try to drag items. Seems somewhat severe? (I don’t have time to troubleshoot it rn). Studio is fully up to date."

## Solution
We must use react-sortable-tree without a managed react-dnd context, and provided it ourself.

Additionally we have to monkeypatchadjust the DragDropMonitor in HTML5Backend to suppress unwanted errors that have no functional impact on the hierarchy, but that leads to annoying error-toasts from Sanity Studio

## Reproduction steps
* Add two hierarchy entries to the structure
* Navigate between the two list items
* Dragging now emits HTML5 backend errors

## Other issues with the plugin
Discovered during work with this issue:

* checkout -> npm install -> npm build fails: latest commit does not compile
* types.d.ts file is imported in createDeskhierarchy. This file is not bundled, so studio get type-errors. It should be renamed to types.ts.
* documentType has a default but is typed as required.
 * Dragging an item into itself emits a error toast and emits error in console. We should just ignore these drag events.
* Sometimes get :"react_devtools_backend.js:3973 Invariant Violation: Expected to find a valid target." errors. Source code from reactdnd indicated that "The invariant message will be stripped in production", so may not be critical, as it does not impact functionality in any other way. Seems to be related with dragging items "quickly".

### Check that the reported error is gone

* Add createDeskHierarchy into any StructureBuilder items list twice ([example](https://github.com/sanity-io/structured-content-2022/blob/main/studio/src/deskStructure.ts#L178))
* Click the first hierarchy, add some stuff, drag and drop
* Click the second hierarchy, click back to the first
* Drag and drop should NOT emit a bunch of error-toasts and/or crash the hierarchy input

### Check that the other discovered errors are resolved

* In a hierarchy, on a node with at least one child
* Drag the parent into itself so it show a red drag target. No error toast should appear after dropping.
* Finally, drag items quickly up and down. No error toasts should appear.

### Check typings
* createDeskHierarchy should not emit Type-errors when providing a documentId. documentType should be optional.

### CODE REVIEW

**Comment by stipsan** [comment]
Before I finish up my review I just want to reply to this:
> I just commited this as patch-changes, but since the types will now "work" this could break existing integrations.
> Should this be a breaking change?
When it comes to potential runtime breakages I like to play it safe and bump major. That's because one of the most frustrating things is when updating a minor/patch casues problems to your end users. On the flipside nobody complains when bumping a major **didn't break anything after all**.
In this case it's just relevant to `TS` so it's a buildtime risk, runtime-wise everything is just the same right? That changes the prioritization IMO, it's more important to maximise the adoption rate of this bugfix than to avoid dev annoyance at CI errors.

File: README.md [hunk 1]
```diff
@@ -295,3 +295,21 @@ We're considering adapting this input to support any type of nest-able data, not
...
+
+## Developing
+
+If you want to test a dev-build of this plugin in the studio there is a bit of a workaround.
+npm/yarn link does not integrate correctly with StructureBuilder and schema part registration in
+the Studio.
+
+The most consistent workflow is:
+
+1.Install [yalc](https://github.com/wclr/yalc)
```
Comment (stipsan) [inline README.md:307]: TIL 🤯  I need to switch to this, will save me so much busywork!!

**Review by stipsan — APPROVED** [review]
# Great work!! 🎉 
## I award 🏆 you a **_Pull Request Approval_** and one bonus meme:
image

And please take the suggestions as *Strong opinions loosely held™*. If any of them resonate with you and you feel like applying them, cool! In any case I consider this PR fit to merge 👍

File: src/components/SuppressedDnDManager.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import {createDndContext} from 'react-dnd'
+import {HTML5Backend} from 'react-dnd-html5-backend'
+import {DragDropManager} from 'dnd-core'
+import {DragDropMonitor, Identifier} from 'dnd-core/lib/interfaces'
```
[the remark this answers is not in the document]
Comment (snorrees) [reply src/components/SuppressedDnDManager.tsx:4]: I dont think there is precedence for this in the repo. This is one of those "make it a lint error or let it go" imo.

File: src/components/TreeEditor.tsx [hunk 3]
```diff
@@ -35,87 +44,111 @@ const TreeEditor: React.FC<{
...
-              treeData={localTree}
-              {...getCommonTreeProps({
-                placeholder: {
-                  title: 'Add items from the list below'
-                }
-              })}
-            />
-          </Card>
+      <DndProvider manager={suppressedDnDManager}>
+        <TreeOperationsContext.Provider value={{...operations, allItemsStatus}}>
```
[the remark this answers is not in the document]
Comment (snorrees) [reply src/components/TreeEditor.tsx:70]: Absolutly agree, slipped by, I tried to get all of those.

File: src/components/SuppressedDnDManager.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (stipsan) [reply src/components/SuppressedDnDManager.tsx:4]: Right, in that case let's not introduce `import type` as consistency is better 👍

File: README.md (hunk 1 above)
Comment (stipsan) [reply README.md:307]: You're a lifesaver @robinpyon! 💖

**Comment by snorrees** [comment]
Rebase-merging this and pulishing 1.1.0 once I have the power to do so.

**Pull request merged**